src/components/data-exploration/embedding/Embedding.jsx

```diff
@@ -132,23 +132,36 @@ const Embedding = (props) => {
   }, [focusedExpression]);
 
   useEffect(() => {
-    if (cellSetHierarchy) {
-      const mapping = cellSetHierarchy.reduce((keyToClusterNameMap, rootHierarchy) => {
-        if (rootHierarchy.children.length > 0) {
-          rootHierarchy.children.forEach((child) => {
-            // eslint-disable-next-line no-param-reassign
-            keyToClusterNameMap[child.key] = _.capitalize(rootHierarchy.key);
-          });
-        }
-        return keyToClusterNameMap;
-      }, {});
-      setClusterKeyToNameMap(mapping);
-    }
+    if (selectedCell) {
+      let extras = {};
+      if (focusData.store === 'genes') {
+        const expressionToDispatch = focusedExpression
+          ? focusedExpression.rawExpression.expression[selectedCell] : undefined;
+        extras = { geneName: focusData.key, expression: expressionToDispatch };
+      }
+
+      // getting the root nodes which are of type cellSets
+      const rootClusterNodes = cellSetHierarchy.map(({ key }) => key)
+        .filter((key) => cellSetProperties[key].type === 'cellSets');
+
+      // getting the cluster properties for every cluster that has the cellId
+      const cellProperties = getCellClassProperties(selectedCell, rootClusterNodes, cellSets);
+
+      const prefixedCellSetNames = [];
+      Object.entries(cellProperties).forEach(([key, clusterProperties]) => {
+        clusterProperties.forEach(({ name, parentNodeKey }) => {
+          prefixedCellSetNames.push(`${_.capitalize(parentNodeKey)} : ${name}`);
+        });
+      });
 
-    if (cellSetProperties) {
-      setCellSetClusters(Object.entries(cellSetProperties).filter(([, cellSet]) => cellSet.type === 'cellSets'));
+      cellInfoTooltip.current = {
+        cellSets: prefixedCellSetNames,
+        cellName: selectedCell,
+        componentType: embeddingType,
+        ...extras,
```

Comment:
I would just pass `geneName` and `expression` here instead of wrapping it in `extras`